src/main/java/io/excaliburfrc/robot/subsystems/Climber.java

```diff
@@ -94,6 +106,51 @@ public void closeAngler() {
     anglerPiston.set(DoubleSolenoid.Value.kReverse);
   }
 
+  private void achieveState(TrapezoidProfile.State setpoint) {
+    ElevatorFeedforward ff =
+        anglerPiston.get() == DoubleSolenoid.Value.kForward ? diagonalFF : upFF;
+    leftController.setReference(
+        setpoint.velocity,
+        ControlType.kVelocity,
+        0,
+        ff.calculate(setpoint.velocity),
+        ArbFFUnits.kVoltage);
+    rightController.setReference(
+        setpoint.velocity,
+        ControlType.kVelocity,
+        0,
+        ff.calculate(setpoint.velocity),
+        ArbFFUnits.kVoltage);
+  }
+
+  private Command reachBarCommand(Consumer<TrapezoidProfile.State> toRun) {
+    return new TrapezoidProfileCommand(elevatorProfile, toRun, this);
+  }
+
+  public Command openToFirstCommand() {
+    return reachBarCommand(this::achieveState);
+  }
+
+  public Command openToSecondCommand() {
+    return reachBarCommand(
+        setpoint -> {
+          if (setpoint.position >= HEIGHT_TO_OPEN_PISTON
+              && anglerPiston.get() != DoubleSolenoid.Value.kForward) {
+            anglerPiston.set(DoubleSolenoid.Value.kForward);
+          }
+        });
+  }
+
+  public Command raiseRobotCommand() {
+    return new ParallelCommandGroup(
+        new RunCommand(() -> leftMotor.set(-1), this)
+            .withInterrupt(() -> leftEncoder.getPosition() <= 0.001)
```

Comment:
please extract this value to a constant, and have it a bit larger? we don't want it to overshoot